internal/interconnections/delete.go

```diff
@@ -0,0 +1,37 @@
+package interconnections
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/spf13/cobra"
+)
+
+func (c *Client) Delete() *cobra.Command {
+	var connectionID string
+
+	deleteConnectionCmd := &cobra.Command{
+		Use:   `delete -i <connection_id> `,
+		Short: "Deletes a interconnection.",
+		Long:  "Deletes the specified interconnection.",
+		Example: `  # Deletes the specified interconnection:
+  metal interconnections delete -i 7ec86e23-8dcf-48ed-bd9b-c25c20958277`,
+
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cmd.SilenceUsage = true
+
+			_, _, err := c.Service.DeleteInterconnection(context.Background(), connectionID).Execute()
```

Comment:
Delete should require `--force` or an interactive prompt.

I'll open this as a new issue to keep this feature unblocked as-is.